conda_build/metadata.py

```diff
@@ -269,9 +269,21 @@ def get_contents(meta_path):
     from conda_build.jinja_context import context_processor
 
     path, filename = os.path.split(meta_path)
-    loaders = [jinja2.PackageLoader('conda_build'),
+    loaders = [# search relative to '<conda_root>/Lib/site-packages/conda_build/templates'
+               jinja2.PackageLoader('conda_build'),
+               # search relative to RECIPE_DIR
                jinja2.FileSystemLoader(path)
                ]
+    
+    # search relative to current conda environment directory
+    conda_env_path = os.environ.get('CONDA_ENV_PATH',  # path to current conda environment
```

Comment:
FWIW: I know that Python 2.7 doesn't officially support Visual Studio 2010 or newer. However, this is easily fixed. We need more modern compilers because some extension modules can't be compiled with VS 2008.